hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/viewfs/InodeTree.java

```diff
@@ -377,9 +422,16 @@ private void createLink(final String src, final String target,
         nextInode = newDir;
       }
       if (nextInode.isLink()) {
-        // Error - expected a dir but got a link
-        throw new FileAlreadyExistsException("Path " + nextInode.fullPath +
-            " already exists as link");
+        if (isNestedMountPointSupported) {
+          // nested mount detected, add a new INodeDirLink that wraps existing INodeLink to INodeTree and override existing INodelink
+          INodeDirLink<T> dirLink = new INodeDirLink<T>(nextInode.fullPath, aUgi, (INodeLink<T>) nextInode);
```

Comment:
For creating the dir link, follow the same pattern as what exists in the above if clause of if (nextInode == null) (i.e. addDirLink creates a new dir link and returns the new INodeDirLink). Is there a reason to deviate? also, what shouldn't setInternalDirFs be called here?